base/doc/ltnews33.tex

```diff
@@ -137,6 +139,36 @@ \section{Extending the hook concept to paragraphs}
 
 
 
+\section{Extending the hook concept to commands}
+
+Up to now the hook management covered hooks for a few core areas, such
+as hooks for the \cs{shipout} process or those in the \env{document}
+environment, as well as generic hooks for file loading (helpful for
+patching) and for abitrary environments (executed by \cs{begin} and
+\cs{end}).
+
+This has now been extended to add \hook{/before} and \hook{/after}
+hooks to any (document-level) command---in theory at that is.  In
+practice the new generic \hook{cmd} hooks, especially the
+\hook{cmd/.../after} hooks may fail with commands that are too complex
+to be automatically patched. If that is the case it will be detected
+and an appropriate error is generated. The restrictions are documented
```

Comment:
Doesn't it? I thought I saw the corresponding message detailing why patching doesn't work. Must confess I didn't ry :-)

Reply:
Oh, you are talking about a failed patch there, like what you'd get with `\AddToHook{cmd/filec@ntents/before}{error}`.  That sentence sounds like (to me, that is) it is talking about `cmd/.../after` hooks that can't exist because how the command is implemented, like `cmd/section/after`.